src/gui/shell.cpp

```diff
@@ -725,9 +725,23 @@ void Shell::handleNeovimNotification(const QByteArray &name, const QVariantList&
 			m_mouseHide = variant_not_zero(args.at(1));
 			int val = m_mouseHide ? 1 : 0;
 			m_nvim->api0()->vim_set_var("GuiMousehide", val);
-		} else if (guiEvName == "Close" && args.size() == 1) {
+		} else if (guiEvName == "Close" && (args.size() == 1 || args.size() == 2)) {
 			qDebug() << "Neovim requested a GUI close";
-			emit neovimGuiCloseRequest();
+
+			// Decide which exit status Neovim-qt should exit with.
+			int status;
+			if (args.size() == 1) {
+				// If there are no arguments, then the exit status is zero.
+				status = 0;
+			} if (args.size() == 2) {
+				// Or, if there is an argument, then the exit status is
+				// specified in the argument.
+				bool ok;
+				status = args.at(1).toInt(&ok);
```

Comment:
It might be a good idea to perform some basic command format checking before parsing or `toInt` conversion?

Ex)
```C++
if (opargs.size() < 2
		// || !opargs.at(0).canConvert<???>() <- May or may not be necessary?
		|| !opargs.at(1).canConvert<int>() {
		qWarning() << "Unexpected arguments {????}" << opargs;
		return;
	}
```

Not strictly necessary, since you check `ok`...

Reply:
Now it emits warning message if `ok` is `false`.  If I should use `canConvert()` or something, I'll fix it.

Reply:
Slight preference for the format in the comment above. That pattern is used throughout `shell.cpp`, and keeping consistency makes reading the code easier.

Somewhat pedantic/nitpicky... Feel free to ignore!